### PR TITLE
Add Logging

### LINKED ISSUE #22: Improve Logging
Improve the log output of the applications. In some parts, logs are quite sophisticated (e.g. map download) whereas other parts are silent (e.g. elevation map creation). We can improve logging by adding the following logs: 

- [ ] Summary of passed command-line args
- [ ] Runtime info, as done for the map download
- [ ] Summary at the end. List created files and directory
- [ ] others...

In this process, we can also switch to [Logging facility for Python](https://docs.python.org/3/library/logging.html) right away.

### PR DESCRIPTION
**Pull request by maede97**
This PR adds logging statements instead of print(...).
This PR will close #22.

### CODE REVIEW

**Comment by maede97** [comment]
@wp99cp I am not fully happy with the current implementation. Any thoughts? Maybe we can use a more variety of log-levels to distinguish between debug and info messages?
Any preferences?

**Comment by wp99cp** [comment]
> @wp99cp I am not fully happy with the current implementation. Any thoughts? Maybe we can use a more variety of log-levels to distinguish between debug and info messages? Any preferences?

We can use `debug` level messages for logs only important for developers, e.g. 

```
[INFO] add_to_database at find_swisstopo_name.py:22: Database has 428204 entries.
[INFO] add_to_database at find_swisstopo_name.py:16: Adding file ./res/swissNAMES3D_PLY.csv with name 5 to database.
[INFO] select_waypoints at find_walk_table_points.py:29: Preselection returned 1081 points

[INFO] create_walk_table at walk_table.py:118: 2.9km 1159m ü. M. 1.3h 10:18Uhr (731433, 207526) bei/m Dreggloch
[INFO] create_walk_table at walk_table.py:118: 0.7km 1298m ü. M. 0.4h 10:42Uhr (731926, 207781) bei/m Gamserchessel
[INFO] create_walk_table at walk_table.py:118: 0.6km 1429m ü. M. 0.4h 11:03Uhr (732438, 207700) bei/m Chäsgaden
```

And use a limited number of `info` level messages is also important to the end-user. These messages should be understandable for people without coding experience or deep knowledge of our script/algorithm, e.g.

```
[INFO] Excel file successfully created
[INFO] PDF 1 of 10 successfully downloaded
``` 

And warning for reporting unusual system states, e.g.

```
[WARNING] No Database with "Flurnamen" found. Creation of new Database started, this may take a while.
```

**Comment by maede97** [comment]
@wp99cp You can now change the log level in main.py, line 6. Default is INFO. This is required somehow, since the root logger determines all other logger levels, and we can only set the root logging level once we parsed the arguments, which would be too late, because the other loggers are already set.
If you like it that way, we can keep it.
Otherwise I think we have to add another custom wrapper around the logging.Logger.

Furthermore, errors are only raised if the log level is DEBUG, which means the "usual" user does not have a stracktrace to debug and only the logging error shown.

**Comment by wp99cp** [comment]
> This is required somehow, since the root logger determines all other logger levels, and we can only set the root logging level once we parsed the arguments, which would be too late, because the other loggers are already set.

I see two options to avoid this problem. The main aspect of this issue is that we import the module with the other loggers before parsing the arguments (namely at the top of the main file)

1) Consistently use classes among all files using logger, then instantiate the logger by calling `logging.getLogger(__name__)` in the constructor of the class. 

2) Slacker variant: inline the import of as many as possible modules to a function called once the args are parsed.

Version (2) should work but is against the philosophy of python to put import statements at the top of a file. Therefore I prefer variant (1).

Nevertheless, since this would result in many changes, you can merge the PR and create a separate issue.

File: python_program/automatic_walk_time_tables/find_walk_table_points.py [hunk 1]
```diff
@@ -4,6 +4,13 @@
 from gpxpy.gpx import GPXTrackPoint
 from typing import List, Tuple
 
+# Set up logging
+from . import log_helper
+import logging
+logger = logging.getLogger(__name__)
+handler = logging.StreamHandler()
+handler.setFormatter(log_helper.Formatter())
+logger.addHandler(handler)
```
Comment (wp99cp) [inline python_program/automatic_walk_time_tables/find_walk_table_points.py:13]: Is it really necessary to the STeramHandler and Formater in a very file? Aren't you using the same logger object anyway by calling `logging.getLogger(__name__)`?

**Review by wp99cp — CHANGES_REQUESTED** [review]
Works as expected, thank's for your work.
Can you quickly look at my comment?

File: python_program/automatic_walk_time_tables/find_walk_table_points.py (hunk 1 above)
Comment (maede97) [reply python_program/automatic_walk_time_tables/find_walk_table_points.py]: Done.

**Review by wp99cp — APPROVED** [review]
Look good.

**Pull request merged**